### ISSUE TITLE
Checking If A Location Is From A Fence

### ISSUE THREAD

**Issue by tonespy** [transistorsoft/flutter_background_geolocation#654]
## Your Environment
* Plugin version: ^4.3.0
* Platform: iOS and Android
* OS version: 11
* Device manufacturer / model: Google Pixel 4 / iPhone 11
* Flutter info (`flutter doctor`):
* Plugin config:
```dart <-- Syntax highlighting: DO NOT REMOVE -->
[✓] Flutter (Channel stable, 2.5.1, on macOS 11.6 20G165 darwin-x64, locale en-NG)
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[✓] Xcode - develop for iOS and macOS
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.1)
[✓] VS Code (version 1.60.0)
[✓] Connected device (1 available)
    ! Error: iPhone XS is not connected. Xcode will continue when iPhone XS is connected. (code -13)
    ! Error: iPhone 12 Pro is not connected. Xcode will continue when iPhone 12 Pro is connected. (code -13)
```

## Expected Behavior
I will like to be able to know when a location gotten from 
```
bg.BackgroundGeolocation.onLocation((location) {
});
```
has additional information of the fence it's gotten from. Is there support for this?

I've something like this for monitoring fence enter, exit and still, as described in the documentation, and starting active and passive tracking in the process.
```
BackgroundGeolocation.onGeofence((GeofenceEvent event) {
  print('[geofence] ' + event.toString());
  if (event.action == 'ENTER') {
      // Entering the danger-zone, we want to aggressively track location.
      BackgroundGeolocation.start();
    } else if (event.action == 'EXIT') {
      // Exiting the danger-zone, we resume geofences-only tracking.
      BackgroundGeolocation.startGeofences();
    }
});
```
But, this is limited.

Basically, when it's motion based location within a fence, can we tell if the location is in that fence as an additional property within the [Location Class](https://pub.dev/documentation/flutter_background_geolocation/latest/flt_background_geolocation/Location-class.html)?

**Comment by christocracy** [comment]
Why don't you just set your own state variable `inGeofence` in your `onGeofence` listener?

Then in your `onLocation`:

```dart
if (_inGeofence) {
  // in a geofence
}
```

**Comment by christocracy** [comment]
> boundary instead of Geofence? 

I don't understand what you mean by "boundary".

A geofence *is* a circular "boundary" (or permimeter) with a radius.  Ie: a circle.

**Comment by christocracy** [comment]
Locations don't *come from* a geofence.

A geofence event is simply an event containing the identifier which fired along with the location where it fired.

Regular tracking locations have no  relation to any particular geofence.

**Comment by tonespy** [comment]
<img width="195" alt="S1 and S2 " src="https://user-images.githubusercontent.com/3057485/136698293-ebb23608-146b-4393-94e5-5ac253e92cb4.png">

Let me use this image for describing what I am trying to say. If you look at **S2**, there's a huge assurance the it covers **S1**. In that case, depending on the circular radius will be confusing. As, I will have two fence active at the same time.

**Comment by tonespy** [comment]
> Aside from the `onGeofence` event, Locations don't _come from_ a geofence.
> 
> A geofence event is simply an event containing the identifier which fired along with the location where it fired.
> 
> Regular tracking locations have no relation to any particular geofence.

Yep. This, I've noticed.

**Comment by christocracy** [comment]
The plugin can't do everything for you.  Geofence events and location events have no relationship.  They are monitored through completely separate apis.

There's nothing the plugin can do for you here.  It's up to your own application code.
